### PR TITLE
WIP: Get things working on Julia 0.5

### LINKED ISSUE #59: Tests fail on v0.5
There are multiple failures: 
```
Error During Test
  Test threw an exception of type MethodError
  Expression: size(df[1:2,:]) == (2,2)
  MethodError: broadcast(::DataFramesMeta.##15#16{TestCompositeDataFrames.CompositeDF##349}) is ambiguous. Candidates:
    broadcast(f, x::Number...) at broadcast.jl:16
    broadcast(f::Function, As::DataArrays.PooledDataArray...) at /Users/ranjan/.julia/v0.5/DataArrays/src/broadcast.jl:318
   in nrow at /Users/ranjan/.julia/v0.5/DataFramesMeta/src/compositedataframe.jl:109 [inlined]
   in size(::TestCompositeDataFrames.CompositeDF##349) at /Users/ranjan/.julia/v0.5/DataFrames/src/abstractdataframe/abstractdataframe.jl:214
   in include_from_node1(::String) at ./loading.jl:426
   in macro expansion; at /Users/ranjan/.julia/v0.5/DataFramesMeta/test/runtests.jl:15 [inlined]
   in anonymous at ./<missing>:?
   in include_from_node1(::String) at ./loading.jl:426
   in process_options(::Base.JLOptions) at ./client.jl:262
   in _start() at ./client.jl:318
	FAILED: compositedataframes.jl
```
and 
```
Test Failed
  Expression: xlinq2[[:meanX,:meanY]] == xlinq[[:meanX,:meanY]]
   Evaluated: 3×2 DataFrames.DataFrame
│ Row │ meanX    │ meanY   │
├─────┼──────────┼─────────┤
│ 1   │ 0.537542 │ 5.37542 │
│ 2   │ 0.521357 │ 5.21357 │
│ 3   │ 0.494239 │ 4.94239 │ == 3×2 DataFrames.DataFrame
│ Row │ meanX    │ meanY   │
├─────┼──────────┼─────────┤
│ 1   │ 0.521357 │ 5.21357 │
│ 2   │ 0.494239 │ 4.94239 │
│ 3   │ 0.537542 │ 5.37542 │
	FAILED: linqmacro.jl
ERROR: LoadError: "Tests failed"
 in include_from_node1(::String) at ./loading.jl:426
 in process_options(::Base.JLOptions) at ./client.jl:262
 in _start() at ./client.jl:318
while loading /Users/ranjan/.julia/v0.5/DataFramesMeta/test/runtests.jl, in expression starting on line 27
```

### PR DESCRIPTION
**Pull request by ararslan**
Fixes #59. cc @ranjanan

### CODE REVIEW

**Comment by tshort** [comment]
Thanks for working on this!

File: src/compositedataframe.jl [hunk 1]
```diff
@@ -62,8 +62,8 @@ df = CompositeDataFrame(:MyDF, x = 1:3, y = [2, 1, 2])
 """
 function CompositeDataFrame(columns::Vector{Any},
                             cnames::Vector{Symbol} = gennames(length(columns)),
-                            typename::Symbol = symbol("CompositeDF" * string(gensym())))
-    rowtypename = symbol(string(typename, "Row"))
+                            typename::Symbol = @compat(Symbol("CompositeDF", gensym())))
```
Comment (nalimilan) [inline src/compositedataframe.jl]: I thought `@compat` was only needed for 0.3? Since the package needs 0.4, you should be able to get rid of these.

Comment (ararslan) [inline src/compositedataframe.jl]: The `Symbol` constructor on 0.4 doesn't take multiple string arguments like this, which is why I used `@compat`.

**Comment by ararslan** [comment]
Hm, I bet the test failure is related to https://github.com/JuliaLang/julia/issues/17426.

**Comment by ararslan** [comment]
`orderby` producing different results on 0.4 and 0.5 seems really unlikely, but the ordering is the cause of the failure. So I'm a little confused about that failure... @tshort do you have any ideas? You're the expert here. 🙇

**Comment by tshort** [comment]
On my windows machine, I can't even get DataFramesMeta to load. I'm getting a weird "ERROR: ArgumentError: DataFramesMeta not found in path."

**Comment by ararslan** [comment]
Well that's troubling. If there's an issue with the package on Windows it might be good to have Windows CI going as well. If you have an AppVeyor account would you be willing to set this package up on it?

**Comment by tshort** [comment]
I have an Appveyor account. I'll see if I can figure out how to set it up.

**Comment by tshort** [comment]
I added it on appveyor.com, but it still needs an appveyor.yml.

**Comment by ararslan** [comment]
Okay, I'll put one in this PR.

**Comment by ararslan** [comment]
Hm. Looks like it's working on AppVeyor so far. Do you have DataFramesMeta installed on the computer you're on? I've definitely made that mistake before.

**Comment by tshort** [comment]
I got a working version on Linux. I think `orderby` is broken. It didn't show up on tests because of an unlucky ordering. With the change in ordering, it's showing the brokenness.

**Comment by tshort** [comment]
Looks like it's just the test that's broken. The linq test should look like this:

```julia
module TestLinqMacro

using Base.Test
using DataArrays, DataFrames
using DataFramesMeta

srand(100)
n = 100
df = DataFrame(a = rand(1:3, n),
               b = ["a","b","c","d"][rand(1:4, n)],
               x = rand(n))

x = @where(df, :a .> 2, :b .!= "c")
x = @transform(x, y = 10 * :x)
x = @by(x, :b, meanX = mean(:x), meanY = mean(:y))
x = @orderby(x, -:meanX)
x = @select(x, var = :b, :meanX, :meanY)

x1 = @linq transform(where(df, :a .> 2, :b .!= "c"), y = 10 * :x)
x1 = @linq by(x1, :b, meanX = mean(:x), meanY = mean(:y))
x1 = @linq select(orderby(x1, -:meanX), var = :b, :meanX, :meanY)

## chaining
xlinq = @linq df  |>
    where(:a .> 2, :b .!= "c")  |>
    transform(y = 10 * :x)  |>
    by(:b, meanX = mean(:x), meanY = mean(:y))  |>
    orderby(-:meanX)  |>
    select(var = :b, :meanX, :meanY)

@test x == x1
@test x == xlinq

xlinq2 = @linq df  |>
    where(:a .> 2, :b .!= "c")  |>
    transform(y = 10 * :x)  |>
    groupby(:b) |>
    orderby(-mean(:x))  |>
    based_on(meanX = mean(:x), meanY = mean(:y))
    
@test xlinq2[[:meanX, :meanY]] == xlinq[[:meanX, :meanY]]

end
```

Some of the `orderby`'s had an extra variable in there.

**Comment by ararslan** [comment]
That did it! Things seem to be working on all platforms on all versions. Thanks, @tshort!

**Comment by tshort** [comment]
Great! Is this good to merge then?

**Comment by ararslan** [comment]
I think so. Thanks again for your help!

**Comment by ararslan** [comment]
Should we tag a new version?

**Comment by tshort** [comment]
Nice job! Thanks for doing this, @ararslan. Yes to tagging a new version.

**Comment by ararslan** [comment]
Okay cool, will do

**Pull request merged**